bindings/rust/s2n-tls/src/connection.rs

```diff
@@ -857,6 +857,13 @@ impl Connection {
         version.try_into()
     }
 
+    pub fn client_hello_version(&self) -> Result<Version, Error> {
```

Comment:
Why is `s2n_connection_get_client_hello_version()` needed instead of [s2n_connection_get_client_protocol_version()](https://github.com/aws/s2n-tls/blob/5d927e63d2a54c6d10470cd7855d72f55eacb77a/api/s2n.h#L3074) or [s2n_client_hello_get_legacy_protocol_version()](https://github.com/aws/s2n-tls/blob/5d927e63d2a54c6d10470cd7855d72f55eacb77a/api/s2n.h#L1622C20-L1622C64)?

Reply:
The fact that this question was asked probably means we need a comment.

I added a comment to [s2n_client_hello_get_raw_message](https://github.com/aws/s2n-tls/blame/main/api/s2n.h#L1462-L1469) about this use case, but it isn't mentioned on [s2n_connection_get_client_protocol_version](https://github.com/aws/s2n-tls/blame/main/api/s2n.h#L3065-L3074) itself. The usage guide also has a [section](https://github.com/aws/s2n-tls/blob/main/docs/usage-guide/topics/ch04-connection.md#protocol-version) on the different protocol versions, but s2n_connection_get_client_protocol_version isn't even mentioned. Clearly we've got some documentation gaps :(

And of course, updating the C documentation doesn't fix Rust :( https://github.com/aws/s2n-tls/issues/4615

Reply:
Oh wait, there is a note elsewhere in the usage guide about it: https://github.com/aws/s2n-tls/blob/933f379bb5aca82ac3fcc9687cdc83dba4c93a4c/docs/usage-guide/topics/ch10-client-hello.md?plain=1#L34-L37

Reply:
Ohh got it okay, that makes sense. Would this be easier to understand if we just made the rust version be `client_hello_is_sslv2()` rather than expose an additional protocol version getter?